### PR TITLE
Speed up smoke test

### PR DESCRIPTION
**Pull request by Raneet10**
# Description

This PR continuously monitors state sync and checkpoints in the CI smoke tests as initially we were waiting for exactly 10 minutes for them to go through.

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have added at least 2 reviewer or the whole pos-v1 team
- [x] I have added sufficient documentation in code
- [x] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

## Testing

- [ ] I have added unit tests
- [ ] I have added tests to CI
- [ ] I have tested this code manually on local environment
- [ ] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai
- [ ] I have created new e2e tests into express-cli

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -154,7 +154,7 @@ jobs:
             cd matic-cli/devnet/code/contracts
             npm run truffle exec scripts/deposit.js -- --network development $(jq -r .root.tokens.MaticToken contractAddresses.json) 100000000000000000000
             cd -
-            bash bor/integration-tests/smoke_test.sh
+            timeout 10m bash bor/integration-tests/smoke_test.sh
```
Comment (marcello33) [inline .github/workflows/ci.yml:157]: I'd increase the potential timeout to 20m just to be sure, as I've seen myself sometime this getting up to roughly 18mins. It won't anyway have an impact when we actively check for completion

File: integration-tests/smoke_test.sh [hunk 2]
```diff
@@ -3,38 +3,40 @@ set -e
...
+    else
+        echo "Found checkpoint ID:" $checkpointID
+        checkpointFound="true"
+    fi
+
+    if [ $stateSyncFound == "true" ]  && [ $checkpointFound == "true" ]; then
+        break
+    fi    
+
+done
```
Comment (marcello33) [inline integration-tests/smoke_test.sh:41]: If you look at the action logs, you'll se that it is flooded with some pleonastic logs which makes it very hard to follow up.
I'd propose to remove the following logs:
```
echo "Found matic balance on account[0]: " $balance
echo "Checkpoint didn't arrive yet! Waiting..."
echo "Found checkpoint ID:" $checkpointID 
echo "Balance in bor network has not increased. Waiting for state sync..."
```
And only print the final success or final ones, to make it more readable.

File: integration-tests/smoke_test.sh [hunk 3]
```diff
@@ -3,38 +3,40 @@ set -e
...
+        stateSyncFound="true"   
+    fi
+
+    checkpointID=$(curl -sL http://localhost:1317/checkpoints/latest | jq .result.id)
+
+    if [ $checkpointID == "null" ]; then
+        echo "Checkpoint didn't arrive yet! Waiting..."
+    else
+        echo "Found checkpoint ID:" $checkpointID
+        checkpointFound="true"
```
Comment (marcello33) [inline integration-tests/smoke_test.sh:34]: Can we add a timer printing the seconds/minutes needed to get the checkpoint?
I believe we can get good indicators by that

File: integration-tests/smoke_test.sh [hunk 4]
```diff
@@ -3,38 +3,40 @@ set -e
...
+        exit 1
+    fi
+
+    echo "Found matic balance on account[0]: " $balance
+
+    if (( $balance <= $balanceInit )); then
+        echo "Balance in bor network has not increased. Waiting for state sync..."
+    else
+        echo "State Sync occured!"
+        stateSyncFound="true"
```
Comment (marcello33) [inline integration-tests/smoke_test.sh:25]: Can we add a timer printing the seconds/minutes needed to get the stateSync?
I believe we can get good indicators by that

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Raneet10) [reply .github/workflows/ci.yml]: [b2a453b](https://github.com/maticnetwork/bor/pull/583/commits/b2a453b115d1bcb4b4a63b291f8ce9c41224e4f0)

File: integration-tests/smoke_test.sh (hunk 2 above)
Comment (Raneet10) [reply integration-tests/smoke_test.sh:33]: [87c54fa](https://github.com/maticnetwork/bor/pull/583/commits/87c54fa7bee290b1ee204e17e403dfdbcad6ce2c)

File: integration-tests/smoke_test.sh (hunk 3 above)
Comment (Raneet10) [reply integration-tests/smoke_test.sh]: [296a9c4](https://github.com/maticnetwork/bor/pull/583/commits/296a9c4ff516c906c7b7b99d8d58306f55b6b4a7)

Comment (marcello33) [reply integration-tests/smoke_test.sh]: Great!

File: integration-tests/smoke_test.sh (hunk 4 above)
Comment (marcello33) [reply integration-tests/smoke_test.sh]: 🎉

**Review by manav2401 — APPROVED** [review]
